lang/cs/Org.Apache.REEF.Common/metrics/Api/SnapshotRequest.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+namespace Org.Apache.REEF.Common.Metrics.Api
+{
+    /// <summary>
+    /// Class representing the Snapshot request.
+    /// </summary>
+    public sealed class SnapshotRequest
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="builder">Metrics record builder to be used to add metrics to record.</param>
+        /// <param name="fullSnapshot">If true, record even uncanged metrics.</param>
+        public SnapshotRequest(IMetricsRecordBuilder builder, bool fullSnapshot)
```

Comment:
IMutableMetric is public and user might want to extend this interface to write their own metrics class in which case he would need to write what to do when snapshot request comes.

Personally, in my ML algorithms like LBFGS, I might like to make my own UpdateFunction inherit IObserver<SnapshotRequest> interface and directly snapshot iterations, loss function etc. without explicitly creating metrics for them.